purchases/src/main/kotlin/com/revenuecat/purchases/google/usecase/GetBillingConfigUseCase.kt

```diff
@@ -0,0 +1,48 @@
+package com.revenuecat.purchases.google.usecase
+
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.BillingConfig
+import com.android.billingclient.api.GetBillingConfigParams
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.PurchasesErrorCallback
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.strings.BillingStrings
+import com.revenuecat.purchases.strings.OfferingStrings
+import java.util.concurrent.atomic.AtomicBoolean
+
+internal class GetBillingConfigUseCase(
+    val onReceive: (BillingConfig) -> Unit,
+    val onError: PurchasesErrorCallback,
+    val withConnectedClient: (BillingClient.() -> Unit) -> Unit,
+    executeRequestOnUIThread: ((PurchasesError?) -> Unit) -> Unit,
+) : BillingClientUseCase<BillingConfig?>(onError, executeRequestOnUIThread) {
+    override val errorMessage: String
+        get() = "Error getting billing config"
+
+    override fun executeAsync() {
+        withConnectedClient {
+            val hasResponded = AtomicBoolean(false)
+
+            getBillingConfigAsync(GetBillingConfigParams.newBuilder().build()) { result, config ->
+                if (hasResponded.getAndSet(true)) {
```

Comment:
Worth it to change to main thread? we are not doing anything critical so I think it's fine to stay in this thread if it's a background thread

Reply:
Right... in the followup PR we are storing stuff in the cache, but other than that we are just logging things, so I think it's ok to stay in the same thread. Lmk if you think otherwise and I can change it in a followup PR.